### PR TITLE
Split RBAC for config-policy and framework

### PR DESCRIPTION
**Pull request by JustinKuli**
Refs:
 - https://github.com/stolostron/backlog/issues/17918

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

### CODE REVIEW

File: helm-charts/policy/templates/policy_framework_cluster_role.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    chart: {{ template "policy.chart" . }}
+    component: "policy"
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - events
+  - namespaces
```
Comment (mprahl) [inline helm-charts/policy/templates/policy_framework_cluster_role.yaml:24]: What needs to create and delete namespaces?

File: helm-charts/policy/templates/config_policy_service_account.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright Contributors to the Open Cluster Management project
+
+# Licensed Materials - Property of IBM
```
Comment (mprahl) [inline helm-charts/policy/templates/config_policy_service_account.yaml:3]: Could you please remove these IBM headers from new files?

File: helm-charts/policy/templates/policy_framework_cluster_role.yaml (hunk 1 above)
Comment (JustinKuli) [reply helm-charts/policy/templates/policy_framework_cluster_role.yaml:24]: spec-sync and status-sync both have this rule, it looks like status-sync uses it here: https://github.com/stolostron/governance-policy-status-sync/blob/main/main.go#L196

File: helm-charts/policy/templates/config_policy_service_account.yaml (hunk 2 above)
Comment (JustinKuli) [reply helm-charts/policy/templates/config_policy_service_account.yaml:3]: Good idea

File: helm-charts/policy/templates/policy_framework_cluster_role.yaml (hunk 1 above)
Comment (mprahl) [reply helm-charts/policy/templates/policy_framework_cluster_role.yaml:18]: I just assumed they were created on cluster import, but that's handy to know. Thanks!

File: helm-charts/policy/templates/policy_framework_cluster_role.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  verbs:
+  - create
+  - delete
```
Comment (mprahl) [inline helm-charts/policy/templates/policy_framework_cluster_role.yaml:21]: Could you please verify if deleting a namespace is required? I'd feel more comfortable if that wasn't part of the controller's ACLs.

Also being able to delete events raises a red flag.

Comment (ycao56) [reply helm-charts/policy/templates/policy_framework_cluster_role.yaml:21]: Yes, we do need to delete the cluster namespace as this namespace is created and only used by policy framework. Maybe we should limit to just allow deleting the namespace with cluster name.

**Pull request closed without merging**